### PR TITLE
feat: added torchvision namespace with roi_align and nms functions

### PR DESCRIPTION
**Pull request by mohame54**
# PR Description

## Related Issue

Close #

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by AnnaTz** [comment]
When I move the tests to under `test_torch` they run fine 🤔

File: ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# ------------ #
+
+
+# nms
+@handle_frontend_test(
+    fn_tree="torchvision.ops.nms",
+    dts_boxes_scores_iou=_nms_helper(),
+    test_with_out=st.just(False),
+)
+def test_torch_nms(
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py:93]: The name should be "test_torchvision_nms".

File: ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        iou_threshold=iou,
+    )
+
+
+# roi_align
+@handle_frontend_test(
+    inputs=_roi_align_helper(),
+    test_with_out=st.just(False),
+)
+def test_torchvision_roi_align(
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py:121]: In this test you have forgotten several fixtures. Basically you have written this like an ivy function test, not a frontend one. We need to add `fn_tree` and `frontend`, remove `fn_name`, use `test_frontend_function` etc.

**Review by AnnaTz — CHANGES_REQUESTED** [review]
Suggested some corrections that caught my eye. Still need to figure out how to run the tests properly.

File: ivy/functional/frontends/__init__.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 versions = {
     "torch": "2.0.1",
+    "torchvision": "0.15.2.",
     "tensorflow": "2.13.0",
     "numpy": "1.25.2",
     "jax": "0.4.14",
```
Comment (AnnaTz) [inline ivy/functional/frontends/__init__.py:9]: Need to add `"test_torchvision"` to `submodules` and `"test_torchvision": ["torchvision", 27],` to `db_dict` in `run_tests_CLI/synchronize_db.py` and in `run_tests.py`.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/__init__.py (hunk 3 above)
Comment (AnnaTz) [inline ivy/functional/frontends/__init__.py:9]: The fixture error gets resolved by adding `conftest.py` under `test_torchvision`.
Should contain this simple code like the rest of the frontend test setups:
```
import pytest


@pytest.fixture(scope="session")
def frontend():
    return "torchvision"
```

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py (hunk 1 above)
Comment (mohame54) [reply ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py:93]: Oh, i forgot about it.

File: ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py (hunk 2 above)
Comment (mohame54) [reply ivy_tests/test_ivy/test_frontends/test_torchvision/test_ops.py:121]: on it.

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hi, seems like you have changed `docs/demos` which you might have done by mistake. Can you revert that change because we can't merge with that change in place?

**Comment by mohame54** [comment]
> Hi, seems like you have changed `docs/demos` which you might have done by mistake. Can you revert that change because we can't merge with that change in place?

but I didn't change it.

File: ivy/functional/frontends/__init__.py (hunk 3 above)
Comment (AnnaTz) [reply ivy/functional/frontends/__init__.py:9]: You forgot the changes in `run_tests.py`.

File: ivy_tests/test_ivy/test_frontends/test_torchvision/confest.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import pytest
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_torchvision/confest.py:1]: The file should be named `conftest` not `confest`. That's why you still can't run the tests 😅

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torchvision/confest.py (hunk 4 above)
Comment (mohame54) [reply ivy_tests/test_ivy/test_frontends/test_torchvision/confest.py:1]: 😆 😆 🤣

**Pull request closed without merging**